pytorch_lightning/metrics/functional/stat_scores.py

```diff
@@ -0,0 +1,312 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Tuple, Optional
+
+import torch
+from pytorch_lightning.metrics.classification.helpers import _input_format_classification
+
+
+def _del_column(tensor: torch.Tensor, index: int):
+    """ Delete the column at index."""
+
+    return torch.cat([tensor[:, :index], tensor[:, (index + 1) :]], 1)
+
+
+def _stat_scores(
+    preds: torch.Tensor, target: torch.Tensor, reduce: str = "micro"
+) -> Tuple[torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor]:
+    """Calculate the number of tp, fp, tn, fn.
+
+    The shape of the returned tensors depnds on the shape of the inputs
+    and the `reduce` parameter:
+
+        * If inputs are of the shape (N, C), then
+
+            * If reduce is 'micro', the returned tensors are 1 element tensors
+            * If reduce is one of 'macro', 'weighted', 'none' or None, the returned
+              tensors are (C,) 1d tensors
+            * If reduce is 'samples, the returned tensors are 1d (N,) tensors
+
+        * If inputs are of the shape (N, C, X), then
+
+            * If reduce is 'micro', the returned tensors are (N,) 1d tensors
+            * If reduce is one of 'macro', 'weighted', 'none' or None, the returned
+              tensors are (N,C) 2d tensors
+            * If reduce is 'samples, the returned tensors are 1d (N,X) 2d tensors
+
+    Parameters
```

Comment:
This part of the docstring is formatted a bit different that other metrics
```suggestion
    Args:
```

Reply:
Ok, will adjust it